cloud/ec2_fleet.go

```diff
@@ -240,15 +240,15 @@ func (m *ec2FleetManager) GetInstanceStatus(ctx context.Context, h *host.Host) (
 
 	instance, err := m.client.GetInstanceInfo(ctx, h.Id)
 	if err != nil {
+		if ec2err, ok := err.(awserr.Error); ok && ec2err.Code() == EC2ErrorNotFound {
```

Comment:
Is there already a test to verify that `GetInstanceStatus` returns `StatusNonExistent` in this case? If not, can we add one?

Reply:
We didn't have a test. I have added one.